pkg/mux/mux.go

```diff
@@ -105,6 +111,9 @@ func (m *Mux) Subscribe(sub event.Subscribe) *Mux {
 	m.mtx.Lock()
 	defer m.mtx.Unlock()
 
+	// make sure we don't hit the rate limit
+	time.Sleep(rateLimit * time.Millisecond)
```

Comment:
hmmm, thats not ideal for people that just use 1 or 5 subscriptions... takes a long time...

how about something that works also well for the many use cases where people have less than 20 connections?

maybe there is a lib for throttling in golang or something we can build with the mutexes we used elsewhere?